src/AutorestSwift/Models/Complex/Schema.swift

```diff
@@ -124,6 +124,8 @@ class Schema: Codable, LanguageShortcut {
                 fatalError("Type mismatch. Expected constant type but got \(self)")
             }
             swiftType = constant.valueType.swiftType()
+        case AllSchemaTypes.duration:
+            swiftType = "TimeInterval"
```

Comment:
"TimeInterval" is, by definition, always in seconds. Is "duration" intended to always be seconds?

Reply:
From the example, i see, yes TimeInterval is in seconds

Reply:
I mean, is "duration" as spelled out in ModelerFour, always guaranteed to be in seconds?

Reply:
The test server expects something like '**P123DT22H14M12.011S**'

which reads as 123 days 22 hr 14 min 12 sec.


```
      case "duration":
        if (scenario === "valid") {
          //For some reason moment.js doesn't quite get the right time value out (due to what looks like floating point issues)
          //so we have to check for two possible times
          if (value === "**P123DT22H14M12.011S**" || value == "P123DT22H14M12.010999999998603S") {
            coverage['HeaderParameterDurationValid']++;
            res.status(200).end();
            break;
          }
        }
```

Reply:
Change to use 'DateInterval' for 'duration'